framework/include/csg/CSGCellList.h

```diff
@@ -0,0 +1,108 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "CSGCell.h"
+
+namespace CSG
+{
+
+/**
+ * CSGCellList creates a container for CSGCell objects to pass to CSGBase object
+ */
+class CSGCellList
+{
+protected:
+  /**
+   * Default constructor
+   */
+  CSGCellList();
+
+  /**
+   * Destructor
+   */
+  virtual ~CSGCellList() = default;
+
+  /**
+   * @brief Add a Material Cell object to cell list
+   *
+   * @param name unique cell name
+   * @param mat_name material name (TODO: this will eventually be a material object and not just a
+   * name)
```

Comment:
we should discuss this. Seems hard to match moose materials to cells. They map to subdomains, which likely will cover multiple cells, but also there can be more than 1 material per subdomain

Reply:
The materials component of the CSG models is definitely a "down-the-road" implementation detail that is yet to be finalized (or really even discussed in detail). For now, we basically envision that developers will use region IDs, etc to use for material names and then we can provide a connection to the concept of materials later. What string is used here and what region/cell it would apply to will be extremely mesh generator dependent.

If your hesitation is about the TODO note, I can remove that until we have a better sense of how that should be handled.